platform/core/src/ffi/c_interface.rs

```diff
@@ -125,10 +126,39 @@ pub unsafe extern "C" fn exec_rdf_update(update: *const c_char) -> CByteBuffer {
     create_byte_buffer(message_pack_serialize(
         cstring_to_str(&update)
             .map(String::from)
-            .and_then(core::exec_rdf_update),
+            .and_then(|query| bytes_to_string(to_json_bytes(core::exec_rdf_update(query)))),
     ))
 }
 
+/// Executes the given RDF query in the feature RDF store.
+/// Returns Result<String, CoreFailure> as MessagePack value.
+#[no_mangle]
+pub unsafe extern "C" fn exec_feature_rdf_query(update: *const c_char) -> CByteBuffer {
+    create_byte_buffer(message_pack_serialize(
+        cstring_to_str(&update)
+            .map(String::from)
+            .and_then(|query| bytes_to_string(to_json_bytes(core::exec_feature_rdf_query(query)))),
+    ))
+}
+
+/// Executes the given RDF update in the feature store.
+/// Returns Result<Void, CoreFailure> as MessagePack value.
+#[no_mangle]
+pub unsafe extern "C" fn exec_feature_rdf_update(update: *const c_char) -> CByteBuffer {
```

Comment:
Is there a common part of this that can be factored off? Up to the argument, the rest of the code is the same.

Reply:
Too hairy for me tbh, possibly something for later, but it's pretty readable this way and as we're all (maybe except Ghenadie) still rust noobs I like it this way